sql/api/src/main/scala/org/apache/spark/sql/types/StringType.scala

```diff
@@ -44,6 +44,9 @@ class StringType private (val collationId: Int) extends AtomicType with Serializ
   private[sql] def supportsLowercaseEquality: Boolean =
     CollationFactory.fetchCollation(collationId).supportsLowercaseEquality
 
+  private[sql] def isNonCSAICollation: Boolean =
+    CollationFactory.isNonCSAI(collationId)
```

Comment:
```suggestion
    !CollationFactory.isCaseSensitiveAccentInsensitive(collationId)
```
(with respect to the changes suggested above)